camel/benchmarks/nexus.py

```diff
@@ -0,0 +1,481 @@
+        """
+        if force_download:
+            logger.info("Force downloading data.")
+            self.download()
+
+        # Mapping dataset names to their corresponding folder names
+        if dataset_name not in dataset_mapping:
+            raise ValueError(
+                f"Dataset {dataset_name} is not recognized. \
+                 Available datasets: {list(dataset_mapping.keys())}"
+            )
+
+        # Get the directory for the dataset
+        dataset_dir = self.data_dir / dataset_name
+        if not dataset_dir.exists():
+            raise FileNotFoundError(
+                f"The dataset directory for {dataset_name} \
+                does not exist at {dataset_dir}. Please download it first."
+            )
+
+        # Load the dataset files
+        dataset = []
+        if dataset_name == "NVDLibrary-NestedCalls":
+            for file_name in os.listdir(dataset_dir):
+                file_path = dataset_dir / file_name
+                if file_name.endswith(".csv"):
+                    data = pd.read_csv(file_path)
+                    for _, sample in data.iterrows():
+                        dataset.append(
+                            NexusSample(
+                                sample["Input"], "".join(sample["Output"])
+                            )
+                        )
+        else:
+            for file_name in os.listdir(dataset_dir / "data"):
+                file_path = dataset_dir / "data" / file_name
+                if file_name.endswith(".parquet"):
+                    data = pd.read_parquet(file_path)
+                    data.head()
+                    for _, sample in data.iterrows():
+                        if dataset_name == "NVDLibrary":
+                            dataset.append(
+                                NexusSample(
+                                    sample["Input"],
+                                    sample["Output"].replace(
+                                        "r = nvdlib.", ""
+                                    ),
+                                )
+                            )
+                        elif dataset_name == "VirusTotal":
+                            dataset.append(
+                                NexusSample(sample["Input"], sample["Output"])
+                            )
+                        elif dataset_name == "PlacesAPI":
+                            dataset.append(
+                                NexusSample(sample["Input"], sample["Output"])
+                            )
+                        elif dataset_name == "ClimateAPI":
+                            dataset.append(
+                                NexusSample(sample["Input"], sample["Output"])
+                            )
+                        elif dataset_name == "OTX":
+                            dataset.append(
+                                NexusSample(sample["Input"], sample["Output"])
+                            )
```

Comment:
These `elif` branches for different `dataset_name` values are performing the same operation. You could simplify this by reorganizing and merging these branches, which would make the code cleaner and easier to maintain.